store/primary/cache.go

```diff
@@ -218,7 +218,7 @@ func (c *syncCache) rotate() {
 }
 
 func (c *syncCache) internEntry(entry *store.IndexEntry) *store.IndexEntry {
-	k := string(entry.ProvID) + cidToKey(entry.PieceID)
+	k := string(entry.ProviderID) + string(entry.Metadata)
```

Comment:
Quick idea (maybe it makes no sense at this point, feel free to disregard it): what about using the first e.g. 20 bytes of Metadata for the key? Just as a sanity-check in case we start storing a lot of metadata per CID in the system, to prevent from ending up with arbitrarily large keys here.

Reply:
hashing the indexentry to get its cid as a key would be preferable to a prefix, but this seems fine for now

Reply:
If we do not use the entirety of the metadata, and the metadata is opaque, then something with only slightly different metadata (in last few bytes) might not be seen as different when it needs to be.  If keys are getting too big, then instead of using the metadata, we can use a hash of the metadata.

If keys are similar, then any common prefixes in the data get compressed out by the radix tree, so depending on what the data looks like, long keys could take up very little space.